### DISCUSSION TITLE
How do node imports work?

### DISCUSSION THREAD

**Discussion by isConic** [leon-thomm/Ryven#100] [Q&A]
I'm trying to do the following. 
![image](https://user-images.githubusercontent.com/48657113/158198667-6371a79d-ffc2-4d90-bdd6-c365aecb45e1.png)

Within this project structure:   
![image](https://user-images.githubusercontent.com/48657113/158199056-7a6f1378-c780-4aa2-8723-9d52a9298c63.png)

When I import them.. they don't show up in the `Ryven` nodes list.

When I've had this same issue of nodes not showing up when I import them,  it usually meant that I wasn't structuring my nodes the "ryven way".  

Like for example... 
To make any of my nodes work in the past, all nodes are declared within the `nodes.py` files, which were located in a directory with the name you'd expect to show up next to your nodes.  
So if I put my facial recognition node called `face_recognition` in a `nodes.py` file that exists in the `facial` directory, it would show up like this.  
<img width="173" alt="image" src="https://user-images.githubusercontent.com/48657113/158200936-8aa5d279-d98a-4da4-9363-e388e4097557.png">
   
If i tried importing anything else that had nodes but was not called `nodes.py` in that some directory. It wouldn't import.  

So now I'm thinking that breaking things into... little modules and exporting them through `nodes.py` is not the "ryven way".   

What is Ryven doing exactly when I give it a `nodes.py` file to import?

**Comment by leon-thomm** [comment]
The README shows how node packages are defined. The package name is the folder's name, and the folder must contain a file `nodes.py`. Of course you can distribute your nodes into other files, you just need to make sure that you import them correctly in your `nodes.py` and call `export_nodes()` on them (you might want to be careful with relative imports in your example, but that's python stuff and doesn't have anything to do with how Ryven processes node packages).

Ryven simply imports the module `nodes.py` which causes `export_nodes()` to execute, which then registers all provided node classes as nodes in the session. That's basically it.

If the nodes don't show up, there's probably an error which you can see in the console hidden at the bottom of the window.

**Reply by leon-thomm** [reply]
ah btw. calling your node package *nodes* is usually probably rather undesirable :)

**Reply by isConic** [reply]
Thanks. I'll give this a try soon.
